test/hotspot/gtest/opto/test_constant_division.cpp

```diff
@@ -46,62 +48,37 @@ juint random<juint, juint>() {
   return os::random() & mask;
 }
 
+#ifdef __SIZEOF_INT128__
 template <>
 julong random<jlong, julong>() {
-  juint bits = juint(os::random()) & 63 + 1;
+  juint bits = juint(os::random()) % 63 + 1;
```

Comment:
This change (`&` => `%`, and the similar change below) go a long way toward explaining why I couldn't
puzzle out what this function was intended to do.  Note that `&` has lower precedence than `+`, so the
earlier version was masking with 64.  The new version doesn't have that operator precedence mistake,
though I'd prefer the precedence be made explicit using parens.

Reply:
Yes that was my mistake, have added parentheses.